symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/DeprecationLogger.java

```diff
@@ -0,0 +1,28 @@
+package com.symphony.bdk.core.util;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.util.Objects;
+
+@Slf4j
+@API(status = API.Status.INTERNAL)
+/**
+ * Class to log deprecation message except if system property bdk.warning.mode is set to none.
+ */
+public class DeprecationLogger {
```

Comment:
rather than having a specific system property what about relying on the log configuration to enable or disable it?
either we could drop the property and give a meaningful name to the logger
and then you could use 

logging.level.BDK_DEPRECATION=ERROR 

to disable the logs?

(for instance https://www.elastic.co/guide/en/elasticsearch/reference/6.7/logging.html#deprecation-logging)